cadence/freshmint-metadata-views/FreshmintMetadataViews.cdc

```diff
@@ -13,4 +15,49 @@ pub contract FreshmintMetadataViews {
             self.hash = hash
         }
     }
+
+    /// Create an instance of MetadataViews.IPFSFile from the provided string.
+    ///
+    /// If the string is in IPFS directory format, this function splits it
+    /// into the its CID and path.
+    ///
+    pub fun ipfsFile(file: String): MetadataViews.IPFSFile {
+        // Find the first "/" character if one exists
+        let maybeIndex = self.firstIndexInString(file, of: "/")
+
+        if let index = maybeIndex {
+            // The CID is all characters before the "/"
```

Comment:
Isn't CID will be all the characters after the last "/" like /ipfs/QmWc7U4qGeRAEgtsyVyeW2CRVbkHW31nb24jFyks7eA2mF

Reply:
Freshmint doesn't include an `/ipfs/` prefix when adding IPFS files to NFTs. I updated the comment with the two expected formats for more clarity.

In general it's hard to validate IPFS strings entirely in the contract, so I think we need to do validation checks in the client software to catch malformed strings.